hyperspy/misc/array_tools.py

```diff
@@ -125,3 +125,33 @@ def sarray2dict(sarray, dictionary=None):
         dictionary[name] = sarray[name][0] if len(sarray[name]) == 1 \
             else sarray[name]
     return dictionary
+
+
+def dict2sarray(dictionary, sarray=None, dtype=None):
+    """Populates a struct array from a dictionary
+
+    Parameters
+    ----------
+    dictionary: dict
+        If dictionary is not None the content of sarray will be appended to the
+        given dictonary
+    sarray: struct array or None
+        Either sarray or dtype must be given. If sarray is given, it is
+        populated from the dictionary.
+    dtype: None, numpy dtype or dtype list
+        If sarray is None, dtype must be given. If so, a new struct array
+        is created according to the dtype, which is then populated.
+
+    Returns
+    -------
+    Structure array
+
+    """
+    if sarray is None:
+        sarray = np.zeros((1,), dtype=dtype)
+    for name in set(sarray.dtype.names).intersection(set(dictionary.keys())):
+        if len(sarray[name]) == 1:
```

Comment:
Not sure about this. The purpose of the similar code in `sarray2dict`
was to avoid using 1d np.array to store metadata. However, in this case
we are actually saving in the array, so, probably there is not point at
doing the same. Also, given that `sarray = np.zeros((1,),
dtype=dtype)`, `len(sarray[<any field>]` should alway be 1 anyway.
Wouldn't `sarray[name] = dictionary[name]` suffice in this case?
On Sun, 2015-12-06 at 07:34 -0800, Vidar Tonaas Fauske wrote:
> In hyperspy/misc/array_tools.py:
> > +    sarray: struct array or None
> > +        Either sarray or dtype must be given. If sarray is given,
> it is
> > +        populated from the dictionary.
> > +    dtype: None, numpy dtype or dtype list
> > +        If sarray is None, dtype must be given. If so, a new
> struct array
> > +        is created according to the dtype, which is then
> populated.
> > +
> > +    Returns
> > +    -------
> > +    Structure array
> > +
> > +    """
> > +    if sarray is None:
> > +        sarray = np.zeros((1,), dtype=dtype)
> > +    for name in
> set(sarray.dtype.names).intersection(set(dictionary.keys())):
> > +        if len(sarray[name]) == 1:
> If the length is 1, we assign the value (scalar/object) to the dict,
> if not, we assign the entire ndarray. Simply the reverse of what
> happens in sarray2dict.
> —
> Reply to this email directly or view it on GitHub.